src/hooks/useClairScanResults.ts

```diff
@@ -0,0 +1,52 @@
+import * as React from 'react';
+import { useK8sWatchResource } from '@openshift/dynamic-plugin-sdk-utils';
+import { TaskRunGroupVersionKind, TaskRunKind } from '../types';
+import { useWorkspaceInfo } from '../utils/workspace-context-utils';
+
+const SCAN_TASK = 'clair-scan';
+const SCAN_RESULT = 'CLAIR_SCAN_RESULT';
+
+export type ClairScanResult = {
+  vulnerabilities: {
+    critical: number;
+    high: number;
+    medium: number;
+    low: number;
+  };
+};
+
+export const useClairScanResults = (name: string): [ClairScanResult, boolean] => {
+  const { namespace } = useWorkspaceInfo();
+  const [taskRuns, loaded] = useK8sWatchResource<TaskRunKind[]>({
```

Comment:
You can replace this with `useTaskRuns` hook and pass the pipelinerun name to fetch taskruns specific to that pipelinerun.